### DISCUSSION TITLE
LogsDao() no longer available in v 0.23.x - is there an alternative for accessing the log output?

### DISCUSSION THREAD

**Discussion by Gys** [pocketbase/pocketbase#5999] [Q&A] [answered]
In the 0.22.x version it was possible to hook into the logging using `app.LogsDao().BeforeCreateFunc`, as suggested for example in https://github.com/pocketbase/pocketbase/issues/4740

Now in 0.23.x this is no longer possible because LogsDao() is no longer defined. 

In other comments (below) it was indicated that the refactoring might bring an alternative, like hooks?

https://github.com/pocketbase/pocketbase/issues/4740#issuecomment-2053746567
https://github.com/pocketbase/pocketbase/discussions/3613#discussioncomment-7394787

**Comment by ganigeorgiev** [comment, marked as answer]
There is no need for it anymore. You can simply attach to the general [`app.OnModelCreate`](https://pocketbase.io/docs/go-event-hooks/#onmodelcreate) hook. This is for example how the live demo redacts the ip and other user data:

```go
app.OnModelCreate("_logs").BindFunc(func(e *core.ModelEvent) error {
    if log, ok := e.Model.(*core.Log); ok && log.Data != nil {
        if _, ok := log.Data["userIP"]; ok {
            log.Data["userIP"] = "[demo_redact]"
        }
        if _, ok := log.Data["remoteIP"]; ok {
            log.Data["remoteIP"] = "[demo_redact]"
        }
        if _, ok := log.Data["userAgent"]; ok {
            log.Data["userAgent"] = "[demo_redact]"
        }
        if _, ok := log.Data["referer"]; ok {
            log.Data["referer"] = "https://pocketbase.io"
        }
    }

    return e.Next()
})
```

**Reply by Gys** [reply]
Thanks!

Earlier I tried `pb.OnRecordCreateRequest("_logs").BindFunc()` but that did not work. What is the difference?

**Reply by ganigeorgiev** [reply]
`OnRecord*` fires only for records. `OnModel` fires for every model (log, collection, record, etc.)
